pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+    "UCCSD",
+    "MPS",
+    "TTN",
+    "AmplitudeEmbedding",
+    "AngleEmbedding",
+    "BasisEmbedding",
+    "IQPEmbedding",
+    "QAOAEmbedding",
+]
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting. A lookup table is used to check the commutation between the
+    controlled, targetted part of operation 1 with the controlled, targetted part of operation 2. It supports
+    most PennyLane operations that are not CV operations.
+
+    Args:
+        operation1 (.Operation): A first quantum operation.
+        operation2 (.Operation): A second quantum operation.
+
+    Returns:
+         bool: True if the operations commute, False otherwise.
+
+    **Example**
+
+    >>> qml.is_commuting(qml.PauliX(wires=0), qml.PauliZ(wires=0))
+    False
+    """
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-return-statements
+
+    control_base = {
+        "CNOT": "PauliX",
+        "CZ": "PauliZ",
+        "CY": "PauliY",
+        "CSWAP": "SWAP",
+        "Toffoli": "PauliX",
+        "ControlledPhaseShift": "PhaseShift",
+        "CRX": "RX",
+        "CRY": "RY",
+        "CRZ": "RZ",
+        "CRot": "Rot",
+        "MultiControlledX": "PauliX",
+        "ControlledOperation": "ControlledOperation",
+    }
+
+    if operation1.name in not_supported_operations or isinstance(
+        operation1, (qml.operation.CVOperation, qml.operation.Channel)
+    ):
+        raise qml.QuantumFunctionError(f"Operation {operation1.name} not supported.")
+
+    if operation2.name in not_supported_operations or isinstance(
+        operation2, (qml.operation.CVOperation, qml.operation.Channel)
+    ):
+        raise qml.QuantumFunctionError(f"Operation {operation2.name} not supported.")
+
+    if operation1.name == "ControlledOperation" and operation1.control_base == "MultipleTargets":
+        raise qml.QuantumFunctionError(f"{operation1.control_base} controlled is not supported.")
+
+    if operation2.name == "ControlledOperation" and operation2.control_base == "MultipleTargets":
+        raise qml.QuantumFunctionError(f"{operation2.control_base} controlled is not supported.")
+
+    # Simplify the rotations if possible
+    if operation1.name in ["U2", "U3", "Rot", "CRot"]:
+        operation1 = simplify(operation1)
+
+    if operation2.name in ["U2", "U3", "Rot", "CRot"]:
+        operation2 = simplify(operation2)
+
+    # Case 1 operations are disjoints
+    if not intersection(operation1.wires, operation2.wires):
+        return True
+
+    # Two CRot that cannot be simplified
+    if operation1.name == "CRot" and operation2.name == "CRot":
+        return two_non_simplified_crot(operation1, operation2)
+
+    # Parametric operation might implement the identity operator
+    simplify_op_1 = simplify_to_identity(operation1, operation2)
+    if simplify_op_1 is not None:
+        return simplify_op_1
+
+    simplify_op_2 = simplify_to_identity(operation2, operation1)
```

Comment:
@rmoyard I find some of these function names a bit confusing - for example, `simplify_to_identity` implies that the function will check if the operations can be simplified, not that they are also commuting!

Maybe they should be called something like `check_identity_commutation`, `check_rot_commutation`, `check_crot_commutation`, etc?

Reply:
The problem is that we have `barrier` and `wirecut` that do not commute with identity, that's why I had to take both operators as arg. But you are right I could rename it.